mne/io/edf/edf.py

```diff
@@ -546,8 +546,8 @@ def _get_info(fname, stim_channel, eog, misc, exclude, infer_types, preload):
     # physical dimension in µV
     edf_info['offsets'] = (
         edf_info['physical_min'] - edf_info['digital_min'] * edf_info['cal'])
-    del edf_info['physical_min']
-    del edf_info['digital_min']
+    #del edf_info['physical_min']
+    #del edf_info['digital_min']
```

Comment:
Please revert, we don't want to preserve these

https://github.com/mne-tools/mne-python/issues/8349

Reply:
why not preserve, how else are you going to get the info of physical minimum and digital minimum? (they are not always symmetric to the maximums)? Is this info retained in any other way.
This info is needed to for example to reconstruct an exported EDF header of a read-in one.

Reply:
Please read through #8349, reasons not to preserve in the private `edf_info` are given there

Reply:
... I also do not see the logic in keeping physical_max but not physical_min info, either all go or all stay, why only keep some fields but not others. physical_max does not make any sense to keep without keeping physical_min ... then this could transition to a more stable version.
I have read the conversation you mentioned, but am I missing the point?

Reply:
>  why only keep some fields but not others

The logic of what gets kept in this private `edf_info` is based on / motivated by keeping the minimum set of stuff MNE-Python needs to read data properly.

> I have read the conversation you mentioned, but am I missing the point?

The main point is that nobody (other than MNE-Python internally) should rely on any private attributes, including those in `edf_info`. Other libraries and/or end users should not use them or care about what they are. So when you say:

>  how else are you going to get the info of physical minimum and digital minimum

... my response based on the above logic is "we do not need them internally for MNE-Python, and end users should not try to get them from MNE-Python". If end users really want these, we need to decide on a public API for them somehow rather than keep them in `edf_info`. Does that make sense?

Reply:
I can see the point of keeping things lean and not wanting to give any potentially unstable info for people to rely on, but then you should also delete the physical and digital minimum and some other info, no? Otherwise this is even more confusing (I am not the first to be confused). Why have this extra info at all then, this point I do not get.

I think it is fine to have this info as "extra" and let people decide if to use it but only rely on that (as [cbrnr](https://github.com/cbrnr) also mentioned in https://github.com/mne-tools/mne-python/issues/8349).

Keeping header derived info as long as sensible (e.g. when manipulating a raw this extra info will be deleted).
Also I  find it hard to come up with a practical example in which this might confuse/mislead someone into using/relying on this info (as also other projects, e.g. FieldTrip keep the parsed header info stored in the data structure as header info).

It might even trick people into thinking that all they need is the physical and digital minimum and want to reconstruct something not reliable l from that fewer info?

For example for me that means I have to maintain another a fork with this info being kept. This is because indeed there is inflexibility in other methods to read edf files and the header info with other means, and that would be the best solution to get at that, at least until there is a more dedicated interface for that. For the case of reading zmax edfs (which do not adhere perfectly to the EDF standard and breaks with pyedflib etc.). 

All in all it would mean a lot of extra coding to get at full header info for the time being by, just so to keep an inconsistent standard from what I can see so far. 

I think this is all that remains to be said from my view. I hope this is convincing enough.
I am newer to this project so have to trust your experience. Thus I am happy to revert to having this info still deleted, so as not have to argue on this further. 
Thanks already for taking the effort in raising this issue again.

Reply:
> but then you should also delete the physical and digital minimum and some other info, no?

No, because this idea (and even this discussion on some level) is what I'm trying to avoid. It's private, so we shouldn't have to be careful about how it appears to anyone external, as users shouldn't ever look at it. If they do, they are totally on their own and should expect breakage, potentially of the silent and dangerous form.

> Until there is a public facing info on this, why not keep it with all the other private info until this public interface is ready?

What you are asking for essentially amounts to us to treatinv this private attribute in a public way -- i.e., to keep it stable so that you can use it 
for a while (until we make a different stable interface). This defeats the purpose of it being private, so to me is a non starter.

Did you try using pyedf or some other EDF-related package to get this metadata? To me that is the best/stable way forward. We could consider making our stuff public, but the other EDF packages are probably better suited to maintain this stuff.